### ISSUE TITLE
Do not export `useTheme` hook

### ISSUE THREAD

**Issue by miroslavstastny** [microsoft/fluentui#20482]
Do not export `useTheme` hook in `react-components`

`useTheme` hook currently returns raw token values. ~~For encapsulation purposes, it should return CSS variables instead.~~

There used to be two different use cases for the `useTheme` hook:
1. `FluentProvider` uses that hook internally to merge themes in a nested provider. This use case requires the raw token values.
2. partner application needs to use the theme tokens.

The first use case is still valid.

For the second use case, the hook is no longer necessary after #20900 is done.
The application can either import the tokens directly from the theme - those will reference the CSS variables injected by `FluentProvider` - or in case of SASS use the SASS tokens implemented in #20808 which use the same CSS variables.

**Comment by miroslavstastny** [comment]
-> @microsoft/teams-prg

**Comment by layershifter** [comment]
TLDR; There is a draft RFC that makes this hook ~~useless~~ non-meaningful: #20651.

---

Proposal is too remove functions from `makeStyles()` calls, tokens will be import directly as CSS variables from `@fluentui/react-theme` ➡️ it comes non-sense as hook's code will be:

```tsx
import { token } from `@fluentui/react-theme`;

function useTheme() {
  return tokens; // 🙉 Just use `tokens`, Luke!
}
```

**Comment by miroslavstastny** [comment]
@layershifter - `useTheme` is still used internally in `FluentProvider` so it cannot be removed.
Should we stop exporting the hook?

**Comment by layershifter** [comment]
> @layershifter - `useTheme` is still used internally in `FluentProvider` so it cannot be removed. Should we stop exporting the hook?

Yes, let's do not export it.
